rasa/core/actions/forms.py

```diff
@@ -317,13 +317,26 @@ def extract_other_slots(
 
         return slot_values
 
+    def get_slot_to_fill(self, tracker: "DialogueStateTracker") -> Optional[str]:
+        """Get the slot to be filled.
+
+        When switching to another form, the requested slot setting is still from the
+        previous form and must be ignored.
+
+        Returns:
+        The slot name or None
+        """
+        return (
+            tracker.get_slot(REQUESTED_SLOT)
+            if tracker.active_loop_name == self.name()
+            else None
+        )
+
     def extract_requested_slot(
         self, tracker: "DialogueStateTracker", domain: Domain
     ) -> Dict[Text, Any]:
-        """Extract the value of requested slot from a user input
-        else return `None`.
-        """
-        slot_to_fill = tracker.get_slot(REQUESTED_SLOT)
+        """Extract the value of requested slot from a user input else return `None`."""
```

Comment:
[We use the descriptive style for docstrings](https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e)
```suggestion
        """Extracts the value of requested slot from a user input else return `None`."""
```